receiver/awsfirehosereceiver/logs_receiver.go

```diff
@@ -35,42 +40,64 @@ var _ firehoseConsumer = (*logsConsumer)(nil)
 func newLogsReceiver(
 	config *Config,
 	set receiver.Settings,
-	unmarshalers map[string]unmarshaler.LogsUnmarshaler,
 	nextConsumer consumer.Logs,
 ) (receiver.Logs, error) {
-	recordType := config.RecordType
-	if recordType == "" {
-		recordType = defaultLogsRecordType
-	}
-	configuredUnmarshaler := unmarshalers[recordType]
-	if configuredUnmarshaler == nil {
-		return nil, errUnrecognizedRecordType
-	}
-
-	mc := &logsConsumer{
-		consumer:    nextConsumer,
-		unmarshaler: configuredUnmarshaler,
+	c := &logsConsumer{
+		config:   config,
+		settings: set,
+		consumer: nextConsumer,
 	}
-
 	return &firehoseReceiver{
 		settings: set,
 		config:   config,
-		consumer: mc,
+		consumer: c,
 	}, nil
 }
 
+func (c *logsConsumer) Start(_ context.Context, host component.Host) error {
+	encoding := c.config.Encoding
+	if encoding == "" {
+		encoding = c.config.RecordType
+		if encoding == "" {
+			encoding = defaultLogsEncoding
+		}
+	}
+	if encoding == cwlog.TypeStr {
+		// TODO: make cwlogs an encoding extension
+		c.unmarshaler = cwlog.NewUnmarshaler(c.settings.Logger)
+	} else {
+		unmarshaler, err := loadEncodingExtension[plog.Unmarshaler](host, encoding, "logs")
+		if err != nil {
+			return err
+		}
+		c.unmarshaler = unmarshaler
+	}
+	return nil
+}
+
 // Consume uses the configured unmarshaler to deserialize the records into a
 // single plog.Logs. It will send the final result
 // to the next consumer.
-func (mc *logsConsumer) Consume(ctx context.Context, records [][]byte, commonAttributes map[string]string) (int, error) {
-	md, err := mc.unmarshaler.Unmarshal(records)
-	if err != nil {
-		return http.StatusBadRequest, err
+func (c *logsConsumer) Consume(ctx context.Context, records [][]byte, commonAttributes map[string]string) (int, error) {
+	logs := plog.NewLogs()
+	for i, record := range records {
+		recordLogs, err := c.unmarshaler.UnmarshalLogs(record)
+		if err != nil {
+			return http.StatusBadRequest, err
+		}
+		if i == 0 {
+			logs = recordLogs
+		} else {
+			recordLogs.CopyTo(logs)
```

Comment:
I don't believe this will behave correctly.  `CopyTo()` will overwrite information in the destination and does not merge entries.  Please ensure there are test cases that include multiple records with combinations of the same and different resource and scope attribute sets to ensure they are correctly produced.

Reply:
:grimacing: thanks for catching that, fixed and added a more comprehensive test.